### ISSUE TITLE
[Bug]: `file.head` happed to break at half or a full byte

### ISSUE THREAD

**Issue by trim21** [beancount/beancount#695]
`file.head` has ability to decode encoding automaticly
```python
    def head_reader(filename):
        with open(filename, 'rb') as file:
            rawdata = file.read(num_bytes)
            file_encoding = encoding or chardet.detect(rawdata)['encoding']
            return rawdata.decode(file_encoding)
```

But I get a encoding error at cornercase:

```text
UnicodeDecodeError: 'gbk' codec can't decode byte 0xc3 in position 8191: incomplete multibyte sequence
```

A multibyte sequence is brak beacuse of `num_bytes=8192`

**Comment by dnicolodi** [comment]
The bug has already been fixed in #646.

**Comment by trim21** [comment]
> The bug has already been fixed in #646.

Sorry, didn't find that issue
